actors/miner/src/lib.rs

```diff
@@ -4090,14 +4090,27 @@ fn handle_proving_deadline(
 
     let state: State = rt.transaction(|state: &mut State, rt| {
         let policy = rt.policy();
-        // Vest locked funds.
-        // This happens first so that any subsequent penalties are taken
-        // from locked vesting funds before funds free this epoch.
-        let newly_vested = state
-            .unlock_vested_funds(rt.store(), rt.curr_epoch())
-            .map_err(|e| e.downcast_default(ExitCode::USR_ILLEGAL_STATE, "failed to vest funds"))?;
-
-        pledge_delta_total -= newly_vested;
+
+        // Vesting rewards for a miner are quantised to every 12 hours and we can determine what those "vesting epochs" are
+        // based on the proving period offset for the Miner.
+        // So, only do the vesting here if the current epoch is a "vesting epoch"
+        let q = QuantSpec {
+            unit: REWARD_VESTING_SPEC.quantization,
+            offset: state.current_proving_period_start(rt.policy(), curr_epoch),
+        };
+
+        if q.quantize_up(curr_epoch) == curr_epoch {
+            // Vest locked funds.
+            // This happens first so that any subsequent penalties are taken
+            // from locked vesting funds before funds free this epoch.
+            let newly_vested =
+                state.unlock_vested_funds(rt.store(), rt.curr_epoch()).map_err(|e| {
+                    e.downcast_default(ExitCode::USR_ILLEGAL_STATE, "failed to vest funds")
+                })?;
+
+            pledge_delta_total -= newly_vested;
+            if pledge_delta_total.is_negative() {}
```

Comment:
Im guessing you meant to delete this.  You probably already figured this out but negative pledge_delta's are normal and expected.

Reply:
@ZenGround0 In the context of the protocol, negative pledge delta means that you don't have enough to cover your pledge requirements and so we will get that from you from future rewards vesting, right  ?

Reply:
A negative pledge _delta_ means a miner's pledge requirement + locked rewards is being reduced, so it actually means some previously-locked tokens are being released. In this case, it's because vesting rewards may have unlocked.